[Feat] : 파일 업로드

Yeon-chae opened the pull request:
## <i>PULL REQUEST</i>

### #️⃣관련 이슈
- close #4 

### 🎋 작업중인 브랜치
- feature/#4

### 💡 작업내용
- S3에 파일 업로드용 presigned URL 받아오는 기능

### 🔑 주요 변경사항
- AWSconfig S3 세팅 추가
- 파일업로드 Dto생성
- FileUploadController,FileUploadService 구현

### 🏞 스크린샷
![image](https://github.com/user-attachments/assets/0e1d0bc9-5fb8-479c-a57f-3fb8f52e2208)
![image](https://github.com/user-attachments/assets/3040844d-339d-4919-b293-2993b5711dc7)
![image](https://github.com/user-attachments/assets/4b9e8698-1d08-4abd-b928-496a5ea10cac)
![image](https://github.com/user-attachments/assets/95503900-64f7-40e4-89aa-2958058fd32b)

Yeon-chae committed:
Merge remote-tracking branch 'origin/feature/#5' into feature/#4

Yeon-chae committed:
[Feat] : AwsConfig S3 설정
- S3관련 설정 추가
- credentials 생성 로직 함수화

Yeon-chae committed:
[Feat] : 파일 업로드용 Dto 작성

Yeon-chae committed:
[Feat] : 파일 업로드 기능 완성
- Controller, Service 클래스 완성
- 반환값은 차후 통일할 예정

khyojun commented:
이후 다른 이슈를 통해서 람다 코드 구성 후 dynamoDB에 들어가는 작업 진행하시는걸까요??

Yeon-chae commented:
> 이후 다른 이슈를 통해서 람다 코드 구성 후 dynamoDB에 들어가는 작업 진행하시는걸까요??
 아무래도 람다 코드를 깃허브에 저장해야하니 이슈를 분리시키고 작업 하려고 합니다!

Yeon-chae committed:
[Feat] : s3 BUCKET 설정 추가

khyojun commented on src/main/java/com/mdt/backend/config/AwsConfig.java:
```diff
@@ -5,42 +5,59 @@
+  }
+
+  @Bean
+  public AmazonS3 amazonS3() {
```
해당 설정 관련하여서 수진님과 s3 서비스를 사용하다보니 conflict 되는 부분이 생길 거 같아 상의하고 수정해야될 거 같아요!

khyojun commented on src/main/resources/application-local.properties:
```diff
@@ -1,3 +1,4 @@
 spring.cloud.aws.credentials.access-key=${AWS_ACCESS_KEY}
 spring.cloud.aws.credentials.secret-key=${AWS_SECRET_KEY}
 spring.cloud.aws.region.static=${REGION}
+spring.cloud.aws.s3.bucket=${cloud.s3.bucket}
```
해당 설정 관련하여서 수진님과 s3 서비스를 사용하다보니 conflict 되는 부분이 생길 거 같아 상의하고 수정해야될 거 같아요!

khyojun reviewed (commented):
s3 관련하여서 내일이라도 수진님과 얘기한 후 일치시키는 과정을 거쳐야겠네요!

Yeon-chae committed:
[Feat] : 파일 업로드 api 반환 메시지 추가

Yeon-chae committed:
Merge remote-tracking branch 'origin/develop' into feature/#4

Yeon-chae committed:
Merge remote-tracking branch 'origin/develop' into feature/#4

# Conflicts:
#	src/main/java/com/mdt/backend/config/AwsConfig.java
#	src/main/resources/application-local.properties

Yeon-chae committed:
[Feat] : 업로드&다운로드 코드 통합
- S3Service로 업로드 로직 이동
- FileService 삭제
- presignedUrl 만료시간 설정 함수화
- 버킷 설정 통일

Yeon-chae committed:
[Chore] : 테스트 데이터 코드 주석화
- AwsConfig의 Primary 어노테이션을 사용하기 위해서 주입 순서 문제 때문에 주석 처리 해두었습니다. 테스트 데이터가 이미 있으니 저 코드는 차후 아예 삭제해도 될 것 같습니다.

khyojun reviewed (approved):
LGTM
